src/datastore/postgres-store.ts

```diff
@@ -2679,6 +2731,150 @@ export class PgDataStore extends (EventEmitter as { new (): DataStoreEventEmitte
     });
   }
 
+  async updateNames(client: ClientBase, bnsName: DbBNSName) {
+    const {
+      name,
+      address,
+      registered_at,
+      expire_block,
+      zonefile_hash,
+      zonefile,
+      namespace_id,
+      latest,
+      tx_id,
+      status,
+      canonical,
+      index_block_hash,
+    } = bnsName;
+    await client.query(`UPDATE names SET latest = $1 WHERE name= $2`, [false, name]);
+
+    await client.query(
+      `
+        INSERT INTO names(
+          name, address, registered_at, expire_block, zonefile_hash, zonefile, namespace_id, latest, tx_id, status, canonical, index_block_hash
+        ) values($1, $2, $3, $4, $5, $6, $7, $8,$9, $10, $11, $12)
+        `,
+      [
+        name,
+        address,
+        registered_at,
+        expire_block,
+        zonefile_hash,
+        zonefile,
+        namespace_id,
+        latest,
+        tx_id,
+        status,
+        canonical,
+        index_block_hash,
+      ]
+    );
+  }
+
+  async updateNamespaces(client: ClientBase, bnsNamespace: DbBNSNamespace) {
+    const {
+      namespace_id,
+      launched_at,
+      address,
+      reveal_block,
+      ready_block,
+      buckets,
+      base,
+      coeff,
+      nonalpha_discount,
+      no_vowel_discount,
+      lifetime,
+      status,
+      latest,
+      tx_id,
+      canonical,
+      index_block_hash,
+    } = bnsNamespace;
+    await client.query(`UPDATE namespaces SET latest = $1 WHERE namespace_id= $2`, [
+      false,
+      namespace_id,
+    ]);
+
+    await client.query(
+      `
+        INSERT INTO namespaces(
+          namespace_id, launched_at, address, reveal_block, ready_block, buckets,
+          base,coeff,nonalpha_discount,no_vowel_discount,lifetime,status,latest,
+          tx_id, canonical, index_block_hash
+        ) values($1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11, $12, $13, $14, $15, $16)
+        `,
+      [
+        namespace_id,
+        launched_at,
+        address,
+        reveal_block,
+        ready_block,
+        buckets,
+        base,
+        coeff,
+        nonalpha_discount,
+        no_vowel_discount,
+        lifetime,
+        status,
+        latest,
+        tx_id,
+        canonical,
+        index_block_hash,
+      ]
+    );
+  }
+
+  async getNamespaceList() {
```

Comment:
This `SELECT` and the couple additional below need a `WHERE canonical = true`. We can just add a `// TODO: fix canonical query` comment for now to merge